### PR TITLE
FEAT: use folder-based flows

### PR DESCRIPTION
**Pull request by anish-mudaraddi**
### Description:
This PR changes the way we use argocd to use folder-based flows
We've been successfully testing this flow on this feature branch so it is ready to go

### Special Notes:

---

### Submitter:

Have you:

* [ ] Labelled this PR, e.g. `bug`, `deployment`, `enhancement` ...etc.

- A `deployment` can be reviewed, and merged, by a single reviewer.
- It can only be used to deploy, change, or remove clusters based on existing patterns for staging.
- Anything involving prod, or production facing services must use the normal 2 person review.
- All other PR types require the usual PR process (e.g. 2 person).

---

### Reviewer

Have you:

* [ ] Verified this PR uses the correct label(s) based on the rules above?
* [ ] Checked if this could affect production (e.g. a global value that's changed without an override)?
* [ ] Tested setting this up, if it's not a deployment, to verify it can be redeployed with any documentation if appropriate?

### CODE REVIEW

File: docs/pics/diagram.drawio
Comment (meoflynn) [inline docs/pics/diagram.drawio:1]: Could we update the diagram to be a PNG file instead of `.drawio`

File: charts/dev/capi-infra/values.yaml [hunk 1]
```diff
@@ -1,4 +1,11 @@
 openstack-cluster:
+  kubernetesVersion: "1.28.7"
+  machineImage: "capi-ubuntu-2004-kube-v1.28.7-2024-03-01"
+
+  # The PEM-encoded CA certificate for openstack.stfc.ac.uk
+  # this expires 2023-12-05T23:59:59Z (UTC)
```
Comment (meoflynn) [inline charts/dev/capi-infra/values.yaml:6]: I don't think this is the right expiry date?

File: charts/staging/capi-infra/values.yaml [hunk 2]
```diff
@@ -0,0 +1,195 @@
+openstack-cluster:
+  kubernetesVersion: "1.28.7"
+  machineImage: "capi-ubuntu-2004-kube-v1.28.7-2024-03-01"
+
+  # The PEM-encoded CA certificate for openstack.stfc.ac.uk
+  # this expires 2023-12-05T23:59:59Z (UTC)
```
Comment (meoflynn) [inline charts/staging/capi-infra/values.yaml:6]: This doesn't seem like the right date

**Review by meoflynn — CHANGES_REQUESTED** [review]
Just did a quick initial review

File: charts/staging/capi-infra/values.yaml (hunk 2 above)
Comment (meoflynn) [reply charts/staging/capi-infra/values.yaml:6]: Discussed in-person, certificate is still valid

File: charts/dev/capi-infra/values.yaml (hunk 1 above)
Comment (meoflynn) [reply charts/dev/capi-infra/values.yaml:6]: Discussed in-person, certificate is still valid

File: docs/pics/diagram.drawio
Comment (anish-mudaraddi) [reply docs/pics/diagram.drawio:1]: this is so you can update the diagrams in https://app.diagrams.net

**Review by meoflynn — APPROVED** [review]

**Review by DavidFair — APPROVED** [review]

**Pull request merged**